lib/autoload/coursemology.rb

```diff
@@ -0,0 +1,7 @@
+module Coursemology
+  class ComponentNotFoundError < StandardError
```

Comment:
Honestly if you want to put it somewhere, putting it in a top-level application module is no different from putting it in the global scope. Find some logical place to put it if you want to have some sort of compartmentalisation.